<h1 align='center'>Python Basic Programming Assignment 4

###  1. Write a Python Program to Find the Factorial of a Number?

In [1]:
num = int(input("Enter the number :"))
c = 1
for i in range(1,num+1):
    c = c*i
print('Facotrial of {} is {}'.format(num,c))

Enter the number :14
Facotrial of 14 is 87178291200


###   2. Write a Python Program to Display the multiplication Table?

In [2]:
#take input as number
num = int(input("Enter the number :"))
print("Table of ",num)
#it
for i in range(1,11):
    print(i*num)

Enter the number :17
Table of  17
17
34
51
68
85
102
119
136
153
170


###  3. Write a Python Program to Print the Fibonacci sequence?


In [3]:
#take length as a input
num = int(input("Length of Fibonacci sequence want : "))

#set f = 0, it is use for incrementing with previous
f = 0

#set first element as 0
f0 = 0 

#set second element as 1
f1 = 1

if num>=1: #check length is greater than 1 or not
    for i in range(num):  #iterate till the end of length provided
            print(f0,end=' ') #print f0
            
            #performing swapping
            f = f0 + f1 
            f0 = f1
            f1 = f
else:
    print("Element list must be greater then 0")

Length of Fibonacci sequence want : 13
0 1 1 2 3 5 8 13 21 34 55 89 144 

###  4. Write a Python Program to Check Armstrong Number?


- Armstrong number : 

            - sum of power of each digit of a number power equal to number of digit in number if it's equal to original  number then it is armstrong number.
            
            - ex :
                    153 (length is 3):
                        = 1**3 + 5**3 + 3**3 
                        = 1 + 125 + 27
                        = 153
               it means it is Armstrong Number
             
             - ex : 
                   1453 (length is 4):
                        = 1**4 + 4**4 + 5**4 + 3**4 
                        = 1 + 126 + 625 + 81
                        = 963
               it means it is not Armstrong Number

In [5]:
#take numper as input
num = int(input("Enter the number : "))

#get length of number
l = len(str(num))

#copy number to other variable
numc = num
a = 0
while(numc!=0):       # check numc is not equal to 0
    r = numc % 10     # get last element of number (remainder after dividing by 10)
    a = a + r**l      # take a power of remainder where power is based on len of number
    numc = numc//10   # get a quotient()

#check number is Armstrong or not
if num==a:
    print('{} is a Armstrong Number'.format(num))
else:
    print('{} is not Armstrong Number'.format(num))

Enter the number : 154
154 is not Armstrong Number


###  5. Write a Python Program to Find Armstrong Number in an Interval?


In [6]:
start = int(input("Enter the start element of Interval :"))
end = int(input("Enter the end element of Interval :"))

print("List of Armstrong number in an Inrerval if {} to {} are:\n".format(start,end))
for num in range(start,end):
    #take numper as input
    #num = int(input("Enter the number : "))

    #get length of number
    l = len(str(num))

    #copy number to other variable
    numc = num
    a = 0
    while(numc!=0):       # check numc is not equal to 0
        r = numc % 10     # get last element of number (remainder after dividing by 10)
        a = a + r**l      # take a power of remainder where power is based on len of number
        numc = numc//10   # get a quotient()

    #check number is Armstrong or not
    if num==a:
        print(num,end=' ')

Enter the start element of Interval :200
Enter the end element of Interval :500
List of Armstrong number in an Inrerval if 200 to 500 are:

370 371 407 

###   6. Write a Python Program to Find the Sum of Natural Numbers?

In [8]:
start = int(input("Enter the start :"))
end = int(input("Enter the end :"))
sum = 0
if start >=0 and end >=start:
    for i in range(start,end+1):
        sum = sum + i
    print('Sum of natural number between and include {} and {} is {}'.format(start,end,sum))
else:
    print('start is must be greater than 0 and end must be greater than start')

Enter the start :1
Enter the end :10
Sum of natural number between and include 1 and 10 is 55


<h3 align='right'>Thank you!